src/glue.rkt

```diff
@@ -138,16 +139,21 @@
 
 (define (taylor-alt altn loc)
   ; BEWARE WHEN EDITING: the free variables of an expression can be null
-  (for/list ([transform transforms-to-try])
-    (match transform
-      [(list name f finv)
-       (alt-event
-        (location-do loc (alt-program altn)
-                     (λ (expr) (let ([fv (free-variables expr)])
-                                 (if (null? fv) expr
-                                     (approximate expr fv #:transform (map (const (cons f finv)) fv))))))
-        `(taylor ,name ,loc)
-        (list altn))])))
+  (define expr (location-get loc (alt-program altn)))
+  (match (type-of expr (for/hash ([var (free-variables expr)]) (values var 'real)))
+    ['real
+      (for/list ([transform transforms-to-try])
+        (match transform
+        [(list name f finv)
+        (alt-event
+          (location-do loc (alt-program altn)
+                       (λ (expr) (let ([fv (free-variables expr)])
+                                      (if (null? fv) expr
+                                          (approximate expr fv #:transform (map (const (cons f finv)) fv))))))
+          `(taylor ,name ,loc)
+          (list altn))]))]
+    ['complex
+      (list altn)]))
```

Comment:
Here, the problem will go away if we support `pow` for complex base / real power, right? Not saying we should, just trying to get a sense of why we add this branch.